ppc64le offload build broken by 8437b7f5584765ad4f7885500647657714930fbb

@jhuber6 This commit 8437b7f5584765ad4f7885500647657714930fbb ( https://github.com/llvm/llvm-project/pull/131205) broke the build on ppc64le.  You can reproduce by configuring with:
```
cmake -G Ninja -B build -S llvm -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD=Native -DLLVM_ENABLE_PROJECTS=clang -DLLVM_ENABLE_RUNTIMES="offload;openmp;compiler-rt" -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang
```

And running:

```
ninja -C build runtimes
```
See attached build logs for details.  Note we aren't seeing the same failures on configurations where llvm is built with `-DPPC_LINUX_DEFAULT_IEEELONGDOUBLE:BOOL=ON`.

This log is confusing it me, one of the errors says this, but it was fixed ages ago. Is this a really old build?
```
/root/llvm-project/llvm/cmake/modules/../../../libc/src/string/memory_utils/inline_memset.h:34:2: error: "Unsupported architecture"
   34 | #error "Unsupported architecture"
```

--- a/libc/src/stdio/printf_core/core_structs.h
+++ b/libc/src/stdio/printf_core/core_structs.h
@@ -55,8 +55,13 @@ struct FormatSection {
   int min_width = 0;
   int precision = -1;
 
-  // Needs to be large enough to hold a long double.
+  // Needs to be large enough to hold a long double. Special case handling for
+  // the PowerPC double double type because it has no FPBits interface.
+#ifdef LIBC_TYPES_LONG_DOUBLE_IS_DOUBLE_DOUBLE
+  Uint128 conv_val_raw;
+#else
   fputil::FPBits<long double>::StorageType conv_val_raw;
+#endif // LIBC_TYPES_LONG_DOUBLE_IS_DOUBLE_DOUBLE
   void *conv_val_ptr;
 
   char conv_name;

